lib/components/base-components/NormalComponent.ts

```diff
@@ -375,6 +395,12 @@ export class NormalComponent<
       rotation: props.pcbRotation ?? 0,
       source_component_id: this.source_component_id!,
     })
+    if (this._footprintError) {
+      const source_component = db.pcb_missing_footprint_error.insert(
+        props._footprintError,
+      )
+      this.source_component_id = source_component.source_component_id
+    }
```

Comment:
this doesn't make sense, `pcb_missing_footprint_error.insert` returns an error object not a source_component, you shouldn't be setting that